modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleFreeContainer.java

```diff
@@ -0,0 +1,221 @@
+package org.testcontainers.containers;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Future;
+
+/**
+ * Testcontainers implementation for Oracle.
+ * <p>
+ * Supported image: {@code container-registry.oracle.com/database/free}
+ * <p>
+ * Exposed ports: 1521
+ * <p>
+ * MIT-licensesd by Bernd Eckenfels, based on OracleContainer from oracle-xe.
+ */
+public class OracleFreeContainer extends JdbcDatabaseContainer<OracleFreeContainer> {
+
+    public static final String NAME = "oracle";
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("container-registry.oracle.com/database/free");
```

Comment:
Please, take into account [this image](https://hub.docker.com/r/gvenzl/oracle-free) instead. It offers many improvements that are suitable for testing.

Reply:
I would like to stick to the official image, but maybe we can make the existing Oracle Ntainer accept all of them, instead?

Reply:
AFAIK, the official image is heavy and it takes around 3 min to start. That's not the experience we want to provide. If you are willing to do that then you can opt-in by using

```java
DockerImageName image = DockerImageName.parse("container-registry.oracle.com/database/free").asCompatibleSubstituteFor("gvenzl/oracle-free");
new OracleFreeContainer(image);
```